src/main/ruby/core/posix.rb

```diff
@@ -132,8 +143,7 @@ def self.getenv(name)
     if value.nil?
       nil
     else
-      ptr = Rubinius::FFI::Pointer.new(Truffle::Interop.as_pointer(value))
-      ptr.read_string_to_null
+      value
     end
   end
```

Comment:
The entire `getenv` method can be removed now that a return_type of :string does the right thing, and then the attach_function is enough.